viewer/src/queries/tracesData.js

```diff
@@ -0,0 +1,16 @@
+export const fetchTracesData = async (traces) => {
+    if (traces.length === 0) {
+        return {}
+    }
+    const returnJson = {};
+    await Promise.all(
+        traces.map(async (trace) => {
+            //This should use react query to reduce calls
```

Comment:
Future optimization

Reply:
By this you mean just pull in the trace once and then use it it across all references?

Reply:
React query caches the result of a query, so it would only do it once if a trace data is shared.  But that seemed like to much to add to this PR because the trace data queries are "signed" so they can change and I need to understand how they change to be able to use it properly.